docs/networking.md

```diff
@@ -0,0 +1,67 @@
+# Advanced networking configurations
+
+It is possible to deploy this sidecar module to different networking configurations to attend different needs.
+
+For testing and evaluation purposes, it is very common that customers will deploy a public sidecar
+(public load balancer and public ECS tasks), but this is not a recommended approach for a production
+environment. In production, typically customers will deploy an entirely private sidecar (private load 
+balancer and private ECS tasks), but sometimes it is necessary to deploy a public load balancer and
+keep ECS tasks in a private subnet. It is also common that more than 5 ports are exposed in a
+production sidecar requiring special attention to the number of services and tasks that will
+be created for that.
+
+See the following sections how to set up the necessary parameters for each of these scenarios.
+
+All resources outlined below are expected to live in the same VPC, meaning that the parameter
+`vpc_id` will correspond to the ID of the VPC of all subnets used throughout the deployment 
+configuration.
+
+## Opening more than 5 ports
+
+ECS imposes a hard limit of 5 target groups per service. This means that if you need to
+use more than 5 ports, the module will automatically create new service definitions for
+each group of up to 5 ports. Use the following parameters to configure more sidecar ports:
+
+* `sidecar_ports`: the list of ports that will be opened at the load balancer.
+
+If setting `sidecar_ports = [443, 1433, 1521, 3306, 5432, 5439, 9996, 9999, 27017, 27018, 27019, 31010]`
+(12 ports), the module will create 3 service definitions to cover the target groups for ports
+`[443, 1433, 1521, 3306, 5432]`, `[5439, 9996, 9999, 27017, 27018]` and `[27019, 31010]`. This
+means that each service will intantiate at least 1 task with the sidecar container.
+If the variable `ecs_service_desired_count` is set to `2` then a total of `6` ECS
+tasks will be instantiated, or `6` sidecar *instances* will be running.
+
+## Public load balancer and public ECS tasks
+
+To deploy an entirely public sidecar, use the following parameters:
+
+* `subnets`: provide public subnets in the same VPC. These subnets will be used for both the EC2
```

Comment:
probably need to change EC2 to ECS services. there is a couple EC2 references in this doc

Reply:
Addressed in [9a4ba96](https://github.com/cyral-quickstart/quickstart-sidecar-aws-ecs/pull/13/commits/9a4ba9647766fe557121af74f52b7793eb812f92).